### PR TITLE
Add OAuth 2.0 authorization login

### PR DESCRIPTION
**Pull request by wu-clan**
PLR.

### CODE REVIEW

File: deploy/docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -76,28 +76,29 @@ services:
...
-#      - fba_server
-#    command:
-#      - nginx
-#      - -g
-#      - daemon off;
-#    volumes:
-#      - fba_static:/www/fba_server/backend/app/static
-#    networks:
-#      - fba_network
+  #  # If you choose to use fastapi_best_architecture_ui (fba_ui), you can open this annotation
```
Comment (downdawn) [inline deploy/docker-compose/docker-compose.yml:79]: Whether to transfer to the front-end project

File: backend/app/core/conf.py [hunk 2]
```diff
@@ -45,6 +45,10 @@ class Settings(BaseSettings):
     # Env Opera Log
     OPERA_LOG_ENCRYPT_SECRET_KEY: str  # 密钥 os.urandom(32), 需使用 bytes.hex() 方法转换为 str
 
+    # OAuth2
+    OAUTH2_GITHUB_CLIENT_ID: str
```
Comment (downdawn) [inline backend/app/core/conf.py:49]: Whether to add the corresponding source link

**Review by downdawn — APPROVED** [review]

File: deploy/docker-compose/docker-compose.yml (hunk 1 above)
Comment (wu-clan) [reply deploy/docker-compose/docker-compose.yml:79]: Maybe, I'll try.

File: backend/app/services/auth_service.py [hunk 3]
```diff
@@ -1,50 +1,45 @@
...
-            # 获取最新用户信息
-            user = await user_dao.get(db, current_user.id)
-            # 创建token
-            access_token, _ = await jwt.create_access_token(str(user.id), multi_login=user.is_multi_login)
-            return access_token, user
+            access_token, _ = await jwt.create_access_token(
+                str(current_user.id), multi_login=current_user.is_multi_login
+            )
+            await user_dao.update_login_time(db, obj.username)
+            await db.refresh(current_user)
```
[the remark this answers is not in the document]
Comment (wu-clan) [reply backend/app/services/auth_service.py:37]: This is to refresh the login time field, or we can omit it.

File: backend/app/core/conf.py (hunk 2 above)
Comment (wu-clan) [reply backend/app/core/conf.py:49]: Good idea.

**Comment by wu-clan** [comment]
please review.

**Review by downdawn — APPROVED** [review]

**Pull request merged**